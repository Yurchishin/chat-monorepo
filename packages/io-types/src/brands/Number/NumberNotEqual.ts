import * as t from 'io-ts'

export interface INumberNotEqualLengthBrand {
    readonly NumberNotEqualLength: unique symbol
}

export const numberNotEqual = (value: number) => t.brand(
    t.number,
    (n): n is t.Branded<number, INumberNotEqualLengthBrand> => n !== value,
    'NumberNotEqualLength',
)
