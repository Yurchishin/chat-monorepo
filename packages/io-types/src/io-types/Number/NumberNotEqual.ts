import * as t from 'io-ts'
import * as E from 'fp-ts/lib/Either'

export interface INumberNotEqualC extends t.Type<number> {}

export const numberNotEqual = (value: number): INumberNotEqualC => new t.Type<number>(
    'NumberNotEqual',
    t.number.is,
    (u, c) =>
        E.either.chain(
            t.number.validate(u, c),
            s => s !== value ? t.success(s) : t.failure(u, c),
        ),
    t.identity,
)
