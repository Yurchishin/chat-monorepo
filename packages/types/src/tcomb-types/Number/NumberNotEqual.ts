import t from 'tcomb-validation'
import R from '@chat/ramda'

export const NumberNotEqual = (value: number) => t.refinement(
    t.Number,
    R.notEquals(value),
    'NumberNotEqual',
)
