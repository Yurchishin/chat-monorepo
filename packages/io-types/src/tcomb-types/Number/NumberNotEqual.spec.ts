import {NumberNotEqual} from './NumberNotEqual'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('NumberNotEqual(5)', () => {
	const notEq5 = NumberNotEqual(5)

	expectTypeSuccess(notEq5, 4)
	expectTypeSuccess(notEq5, 6)
	expectTypeFailure(notEq5, 5)
	expectTypeFailure(notEq5, '1')
})

