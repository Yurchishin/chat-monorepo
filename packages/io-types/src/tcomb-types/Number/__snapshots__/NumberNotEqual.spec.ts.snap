// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberNotEqual(5) expectTypeFailure(1): TYPE_FAILURE::NumberNotEqual(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "1",
      "expected": [Function],
      "message": "Invalid value \\"1\\" supplied to Number",
      "path": Array [],
    },
  ],
  "value": "1",
}
`;

exports[`NumberNotEqual(5) expectTypeFailure(5): TYPE_FAILURE::NumberNotEqual(5) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 5,
      "expected": [Function],
      "message": "Invalid value 5 supplied to NumberNotEqual",
      "path": Array [],
    },
  ],
  "value": 5,
}
`;
